readthedocs/donate/views.py

```diff
@@ -77,6 +77,26 @@ def get_template_names(self):
         return [self.template_name]
 
 
+class PromoDetailView(TemplateView):
+    template_name = 'donate/promo_detail.html'
+
+    def get_context_data(self, **kwargs):
+        promo_slug = kwargs['promo_slug']
+        slugs = promo_slug.split(',')
+        days = int(self.request.GET.get('days', 90))
```

Comment:
Input isn't sanitized here, this should be wrapped for exception handling on `TypeError`